components/common/hooks/useGetTotalUsersNoBots.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect, useState} from 'react';
+
+import {Client4} from 'mattermost-redux/client';
+
+const useGetTotalUsersNoBots = (includeInactive = false): number => {
+    const [userCount, setUserCount] = useState<number>(0);
+
+    const getTotalUsers = async () => {
+        const data = await Client4.getFilteredUsersStats({include_bots: false, include_deleted: includeInactive});
```

Comment:
We try to avoid the pattern for having calling client4 directly in component or hook in this case. cc @hmhealey thoughts

Reply:
I notice there that other hooks that also do the call directly to client4 so I followed that pattern, not sure if there is a reason or block for that maybe from hooks is not possible/easy to use an action?

Reply:
Yes few days back i and @hmhealey noticed quite a few places actually which started using it that way, so bought up the conversation. Yes you are i will unblock it anyways.

Reply:
Good catch, Zubair. This definitely isn't something we should be doing.

The main reason we don't want to call Client4 methods outside of Redux actions is that it bypasses some of the stuff that we rely on Redux middleware for like the standard error handler. It also sort of keeps them all in one sort of layer of the app by keeping them in those.

As for those other places doing this, I'll schedule something to bring this up in either web guild or dev meeting after the holidays. I would've said something about this if I'd noticed it earlier, but since we have so many people working on the web app now, it's hard to keep track of stuff like this that isn't technically wrong but is a bad practice. I definitely struggle with how to deal with this sort of tech debt where there's a bad pattern that's spread quite a bit before I come across it